experimental/ir/zero_test.go

```diff
@@ -0,0 +1,123 @@
+package ir_test
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/bufbuild/protocompile/experimental/id"
+	"github.com/bufbuild/protocompile/experimental/ir"
+)
+
+func TestZero(t *testing.T) {
+	t.Parallel()
+
+	testZeroAny[*ir.File](t)
+	testZeroAny[ir.Import](t) // Import embeds *ir.File
+	testZeroAny[*ir.Imports](t)
+
+	testZeroNode[ir.FeatureSet](t)
+	testZero[ir.Feature](t)
+	testZero[ir.FeatureInfo](t)
+
+	testZeroNode[ir.Member](t)
+	testZeroNode[ir.Extend](t)
+	testZeroNode[ir.Oneof](t)
+	testZeroNode[ir.ReservedRange](t)
+	testZero[ir.ReservedName](t)
+
+	testZero[ir.Service](t)
+	testZero[ir.Method](t)
+
+	testZeroNode[ir.Symbol](t)
+
+	testZeroNode[ir.Type](t)
+
+	testZeroNode[ir.Value](t)
+	testZeroNode[ir.MessageValue](t)
+	testZero[ir.Element](t)
+}
+
+// zeroable is a helper interface to enforce that types implement the IsZero method.
+type zeroable interface {
+	IsZero() bool
+}
+
+// node is a helper interface to enforce [id.Node] types.
+type node[T any] interface {
+	zeroable
+	ID() id.ID[T]
+}
```

Comment:
Wasn't sure if it's entirely necessary to enforce `id.Node`s in this way.
Similarly, I'm not sure if it is necessary to enforce `zeroable` either, since `*ir.File` for example would not conform.